ako-gateway-api/k8s/gateway_controller.go

```diff
@@ -712,3 +728,26 @@ func validateAviConfigMap(obj interface{}) (*corev1.ConfigMap, bool) {
 	}
 	return nil, false
 }
+
+func validateReferreedHTTPRoute(key string, gateway *gatewayv1.Gateway, allowedRoutesAll bool) ([]*gatewayv1.HTTPRoute, error) {
+	namespace := gateway.Namespace
+	if allowedRoutesAll {
+		namespace = metav1.NamespaceAll
+	}
+	hrObj, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().HTTPRoutes(namespace).List(labels.Set(nil).AsSelector())
```

Comment:
Should we look for HTTPRoute only in same namespace or in all namespaces ?

Reply:
We have to look for allowedroutes else we won't re-validate the route in other namespace.

Reply:
I missed reading line 735, my bad

Reply:
changed